File: Src/Icm.Core/Tools.vb
Imports System.Runtime.CompilerServices
Imports System.Text

Namespace Icm

    Public Module Tools


        ''' <summary>
        ''' String join function that accepts a ParamArray of strings.
        ''' </summary>
        ''' <param name="sep"></param>
        ''' <param name="strs"></param>
        ''' <returns></returns>
        ''' <remarks>Ignores null and empty strings.</remarks>
        Public Function JoinStr(ByVal sep As String, ByVal ParamArray strs() As String) As String
            Dim sb As New StringBuilder
            Dim i As Integer = LBound(strs)
            Dim firstOne As Boolean = True
            Do
                If i > UBound(strs) Then
                    Return sb.ToString
                ElseIf String.IsNullOrEmpty(strs(i)) Then
                    i += 1
                Else
                    If firstOne Then
                        sb.Append(strs(i))
                        firstOne = False
                    Else
                        sb.Append(sep & strs(i))
                    End If
                    i += 1
                End If
            Loop
        End Function

    End Module

End Namespace

File: Src/Icm.Core/Tools.cs
using System.Text;

namespace Icm
{

	public static class Tools
	{


		/// <summary>
		/// String join function that accepts a ParamArray of strings.
		/// </summary>
		/// <param name="sep"></param>
		/// <param name="strs"></param>
		/// <returns></returns>
		/// <remarks>Ignores null and empty strings.</remarks>
		public static string JoinStr(string sep, params string[] strs)
		{
			StringBuilder sb = new StringBuilder();
			int i = strs.GetLowerBound(0);
			bool firstOne = true;
			do
			{
			    if (i > strs.GetUpperBound(0)) {
					return sb.ToString();
				}

			    if (string.IsNullOrEmpty(strs[i])) {
			        i += 1;
			    } else {
			        if (firstOne)
                    {
			            sb.Append(strs[i]);
			            firstOne = false;
			        } else {
			            sb.Append(sep + strs[i]);
			        }

			        i += 1;
			    }
			} while (true);
		}
	}
}